### PR TITLE
jas_gene_detail

### PR DESCRIPTION
**Pull request by AlanSimmons**

### CODE REVIEW

**Review by yuanzhou — CHANGES_REQUESTED** [review]

File: src/cells_index/build_index.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+from hubmap_api_py_client.errors import ClientError
+
+logging.basicConfig(format='[%(asctime)s] %(levelname)s in %(module)s:%(lineno)d: %(message)s',
+                            datefmt='%Y-%m-%d %H:%M:%S',
+                            level=logging.INFO)
+logger = logging.getLogger(__name__)
+logger.info(f'Getting information from Cells API...')
+
+# Instantiate hubmap-api-py-client.
+client_url = 'https://cells.dev.hubmapconsortium.org/api/'
```
[the remark this answers is not in the document]
Comment (AlanSimmons) [reply src/cells_index/build_index.py]: Changed. The build_index script is a prototype utility that is not part of the Flask application, but shares the app.cfg file.

File: src/hs_ontology_api/routes/cellsclient.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+        logging.basicConfig(format='[%(asctime)s] %(levelname)s in %(module)s:%(lineno)d: %(message)s',
+                            datefmt='%Y-%m-%d %H:%M:%S',
+                            level=logging.INFO)
+        self.logger = logging.getLogger(__name__)
+
+        #self.logger.info(f'Startup: getting preliminary information from Cells API...')
+
+        # Instantiate hubmap-api-py-client.
+        self.client_url = 'https://cells.dev.hubmapconsortium.org/api/'
```
[the remark this answers is not in the document]
Comment (AlanSimmons) [reply src/hs_ontology_api/routes/cellsclient.py]: Done

File: hs-ontology-api-spec.yaml [hunk 3]
```diff
@@ -340,18 +340,55 @@ paths:
...
-      summary: Return detailed information on genes specified by HGNC identifiers.
-      requestBody:
-        required: true
-        content:
-          application/json:
-            schema:
-              $ref: "#/components/schemas/GeneDetailRequest"
-            example:
-              ids: [60,MMRN1,FANCS]
+  /gene_list:
```
[the remark this answers is not in the document]
Comment (AlanSimmons) [reply hs-ontology-api-spec.yaml:343]: I'll meet you in the middle:

- GET `/genes` - Retrieves a high-level list of genes with parameters for pagination:
   - pages
   - genesperpage
- GET `/gene/<id> - Retrieves detailed information on a specific gene

Calling `/gene` without a parameter returns the error message `The /gene endpoint should specify a single HGNC gene identifier--e.g., /gene/MMRN1.`

File: hs-ontology-api-spec.yaml [hunk 4]
```diff
@@ -340,18 +340,55 @@ paths:
...
+        '200':
+          description: high-level information on genes specified by HGNC identifiers.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/GeneListResponse'
+
+  /gene_detail:
```
[the remark this answers is not in the document]
Comment (AlanSimmons) [reply hs-ontology-api-spec.yaml:380]: I'll meet you in the middle:

- GET `/genes` - Retrieves a high-level list of genes with parameters for pagination:
   - pages
   - genesperpage
- GET `/gene/<id> - Retrieves detailed information on a specific gene

Calling `/gene` without a parameter returns the error message `The /gene endpoint should specify a single HGNC gene identifier--e.g., /gene/MMRN1.`

**Review by yuanzhou — APPROVED** [review]

**Pull request merged**